lib/response.js

```diff
@@ -439,8 +439,8 @@ module.exports = {
     if (this.headerSent) return;
 
     if (2 == arguments.length) {
-      if (Array.isArray(val)) val = val.map(String);
-      else val = String(val);
+      // maybe we can drop support of undefined in the next major version
+      if (val === undefined) val = 'undefined';
```

Comment:
other type of values will be stringify in `res.setHeader`

Reply:
res.getHeader will broken change after this change.

Reply:
> Sets a single header value for headers object. If this header already exists in the to-be-sent headers, its value will be replaced. Use an array of strings here to send multiple headers with the same name. Non-string values will be stored without modification. Therefore, request.getHeader() may return non-string values. However, the non-string values will be converted to strings for network transmission.

Reply:
fixed